tests/js/server/dump/dump-test.inc

```diff
@@ -168,7 +235,7 @@
         assertEqual(properties.numberOfShards, 1);
         assertEqual(properties.replicationFactor, 'satellite');
         if (isCluster) {
-          assertEqual(properties.shardingStrategy, 'enterprise-compat');
+          assertEqual(properties.shardingStrategy, 'hash');
```

Comment:
Why are the sharding strategies change here?
I don't like this, because if we change the sharding strategies for existing collections, we may not find data back later...

Reply:
I think in this test it is actually correct, we are testing dump-restore here.
So we restore a collection into a new cluster, and insert the documents via Collection API.
So we are free to switch sharding strategies here.

However you may have a point here that some other code-path is now broken.